lms/services/canvas_api/public.py

```diff
@@ -0,0 +1,236 @@
+"""High level access to Canvas API methods."""
+
+import marshmallow
+from marshmallow import EXCLUDE, Schema, fields, post_load, validate, validates_schema
+
+from lms.services import CanvasAPIError
+from lms.validation import RequestsResponseSchema
+
+
+class _SectionSchema(Schema):
+    """
+    Schema for an individual course section dict.
+
+    These course section dicts appear in various different Canvas API responses.
+    This _SectionSchema class is used as a base class or nested schema by
+    various schemas below for Canvas API responses that contain section dicts.
+    """
+
+    class Meta:
+        unknown = EXCLUDE
+
+    id = fields.Int(required=True)
+    name = fields.String(required=True)
+
+
+class CanvasAPIClient:
```

Comment:
This is the most copy paste of the files, with the difference that we are composed with the underlying client, rather than being one. Little here changed.